### PR TITLE
[MCC-1058568] Remove `werkzeug` Presumption from WSGI Middleware

### PR DESCRIPTION
**Pull request by ejinotti-mdsol**
# [MCC-1058568](https://jira.mdsol.com/browse/MCC-1058568)

### CODE REVIEW

File: mauth_client/middlewares/wsgi.py [hunk 1]
```diff
@@ -60,3 +62,54 @@ def _read_body(self, environ):
...
+
+            if (
+                (scheme == "https" and port != 443)
+                or
+                (scheme != "https" and port != 80)
+            ):
+                url_parts.append(f":{port}")
+
+        url_parts.append(quote(environ.get("SCRIPT_NAME", "")))
+        url_parts.append(quote(environ.get("PATH_INFO"), ""))
```
[the remark this answers is not in the document]
Comment (ykitamura-mdsol) [reply mauth_client/middlewares/wsgi.py:105]: That sounds about right but it reminds me of [this comment in mauth-client-ruby](https://github.com/mdsol/mauth-client-ruby/blob/7975d2d5370e58ada7996ae6548bbbde081dee0a/lib/mauth/client/local_authenticator.rb#L16-L20) so we may face some edge cases anyhow...

> We are in an unfortunate situation in which Euresource is percent-encoding parts of paths, but not
        all of them.  In particular, Euresource is percent-encoding all special characters save for '/'.
        Also, unfortunately, Nginx unencodes URIs before sending them off to served applications, though
        other web servers (particularly those we typically use for local testing) do not.  The various forms
        of the expected string to sign are meant to cover the main cases.

[the remark this answers is not in the document]
Comment (ejinotti-mdsol) [reply mauth_client/middlewares/wsgi.py]: updated to use the safe chars like werkzeug: [6a77a20](https://github.com/mdsol/mauth-client-python/pull/43/commits/6a77a2008a3e0aed78fb8afa5934c268f6f7d157)

**Comment by ejinotti-mdsol** [comment]
@mdsol/platform-scalability

**Review by masongup-mdsol — COMMENTED** [review]
This looks okay, but shouldn't there be some tests against this URL processing?

**Comment by ejinotti-mdsol** [comment]
@masongup-mdsol yea, that is actually a big chunk of tech debt for this client.. it basically runs the protocol suite against the sign/verify code but not any of the provided middlewares.  Then the middleware tests just stub out the sign/verify code.  I think the Ruby client does similar things.

**Review by ykitamura-mdsol — APPROVED** [review]
looks good to me 👍

**Pull request merged**